### PR TITLE
Add a small tutorial to use Ansible to set up Wildfly

### PR DESCRIPTION
**Pull request by rpelisse**
First time I'm blogging on the Wildfly website, so feel free to let me know what I did wrong :)

### CODE REVIEW

**Comment by rpelisse** [comment]
@bstansberry Friendly reminder :)  (in case you're back from PTO today)

**Comment by darranl** [comment]
FYI plenty of engineers can provide review, not necessarily Brian - if you can get some reviews in should be easier for one of the mergers to click merge.

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+. Install Wildfly on the targeted hosts (here just localhost). This means downloading the archive from this website and decompressing the archive in the appropriate directory (JBOSS_HOME). These tasks are handled by the wildfly_install role supplied by Ansible collection for Wildfly.
+. Create the configuration files to run several instances of Wildfly. Because we’re running multiple instances on a single host, you also need to ensure that each instance has its own subdirectories and set of ports, so that the instances can coexist and communicate. Fortunately, this functionality is provided by a role within the Ansible collection called wildfly_systemd.
+
+=== Ansible playbook to install Wildfly
+
+Here is the playbook we'll use to deploy our clusters. Its content is relatively self-explanitory, at least if you are somewhat familiar with the Ansible syntax.
+
+[source, yml]
+---
+---
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:76]: Duplicate; should be removed.

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+In short, this playbook uses the Ansible collection for Wildfly to, first, install the appserver by using the wildfly_install role. This will download all the artifacts, create the required system groups and users, install dependency (unzip) and so on. At the end of its execution, all the tidbits required to run Wildfly on the target host are installed, but the server is not yet running. That’s what happening in the next step.
+
+In the tasks section of the playbook, we then call on another role provided by the collection: wildfly_systemd. This role will take care of integrating Wildfly, as a regular system service, into the service manager. Here, we use a loop to ensure that we create not one, but three different services. Each one will have the same configuration (standalone-ha.xml) but runs on different ports, using a different set of directories to store its data.
+
+=== Run the playbook!
+
+Now, let’s run our Ansible playbook and observe its output:
+
+[source, bash]
+---
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:136]: 4 dashes

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+TODO
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:137]: Interesting output! ;-)

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+---
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:138]: 4 dashes

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+Note that the playbook is not that long, but it does a lot for us. It performs almost 100 different tasks! Starting by automatically installs the dependencies, including the JVM required by Wildfly, along with downloading its binaries. And the wildfly_systemd role sets does even more, effortlessly setting up three distinct services, each with its own set of ports and directory layout to store instance-specific data.
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:140]: s/installs/installing

Either s/role sets does/role sets do/g or s/role sets/role set/g

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+Even better, the Wildfly installation is NOT duplicated. All of the binaries live under the /opt/wildfly-27.0.1 directory, but all the data files of each instance are stored in separate folders. This means that we just need to update the binaries, once, and then restart the instances, to deploy a patch or upgrade to a new version of Wildfly.
+
+On top of everything, we configured the instances to use the standalone-ha.xml configuration as the baseline, so they are already set up for clustering.
+
+=== Check that everything worked as expected
+
+The easiest way to confirm that the playbook did indeed install Wildfly and started three instances of the appserver is to use the systemctl command to check the associate services state:
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:148]: The : implies there will be an example block.

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+=== Deploy an application to the JBoss EAP cluster
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:150]: s/JBoss EAP/WildFly/g

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+Now, our three Wildfly are running, but the cluster has yet to form. Indeed, with no apps there is no reason for the cluster to exists. Let's modify our Ansible playbook to deploy a simple application to all instances, this will allow us to check that the cluster is working as expected. To achieve this, we'll leverage another role provided by the Wildfly collection: wildfly_utils.
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:152]: s/exists/exist
s/instances,/instances;/g

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+
+In our case, we will use the jboss_cli.yml task file, which encapsulates the running of JBoss command-line interface (CLI) queries:
+[source, yaml]
+---
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:156]: 4 dashes

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+          tasks_from: jboss_cli.yml
+        vars:
+          jboss_home: "{{ wildfly_home }}"
+          query: "'deploy --force {{ wildfly_install_workdir }}/{{ app.name }}'"
+          jboss_cli_controller_port: "{{ item }}"
+        loop:
+          - 9990
+          - 10090
+          - 10190
+---
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:176]: 4 dashes

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+
+Now, we will once again execute our playbook so that the web application is deployed on all instances. Once the automation completes successfully, the deployment will trigger the formation of the cluster.
+
+=== Verify that Wildfly cluster is running  and the app is deployed
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:180]: s/WildFly/the WildFly/g

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+
+You can verify the cluster formation by looking at the log files of any of the three instances:
+
+[source, bash]
+---
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:185]: 4 dashes

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+...
+…
+
+2022-12-23 15:02:08,252 INFO  [org.infinispan.CLUSTER] (thread-7,ejb,jboss-eap-0) ISPN000094: Received new cluster view for channel ejb: [jboss-eap-0] (3) [jboss-eap-0, jboss-eap-1, jboss-eap-2]
+…
+---
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:191]: 4 dashes

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+
+== Conclusion
+
+Here you go, with a short and simple playbook, we have fully automated the deployment of a Wildfly cluster! This playbook can now be used against one, two, three remote machine or even hundreds of them! I hope this will post will have been informative and that it'll have convinced you to use Ansible to set up your own Wildfly servers!
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:195]: s/Wildfly/WildFly

**Review by bstansberry — CHANGES_REQUESTED** [review]
Thanks, Romain. I've made some copy-editing suggestions along with a rendering fix (repeated a lot.) Plus there are a couple missing example blocks.

Ping me on zulip when ready and I can turn it around pretty fast.

File: _posts/2022-12-21-ansible-wildly.adoc [hunk 15]
```diff
@@ -0,0 +1,473 @@
...
+Now, we will once again execute our playbook so that the web application is deployed on all instances. Once the automation completes successfully, the deployment will trigger the formation of the cluster.
+
+=== Verify that the WildFly cluster is running  and the app is deployed
+
+You can verify the cluster formation by looking at the log files of any of the three instances:
+
+[source, bash]
+----
+...
+…
```
Comment (bstansberry) [inline _posts/2022-12-21-ansible-wildly.adoc:456]: Nit: there are two ... lines here.

**Review by bstansberry — CHANGES_REQUESTED** [review]
Thanks @rpelisse!

There's one item still to update, and I noticed a different minor thing. The blog renders fine now though. :)

Also, please rename the file to 20230-01-10-ansible-wildfly.adoc. The file name doesn't really affect how it appears on the site or the URL, but having the names match or close to the URL is helpful for maintenance, finding content to steal from later, etc.

**Review by bstansberry — APPROVED** [review]

**Pull request merged**